cmd/sniproxy/snipproxy.go

```diff
@@ -197,3 +208,10 @@ func (s *server) dnsResponse(req *dnsmessage.Message) (buf []byte, err error) {
 
 	return resp.Finish()
 }
+
+func (s *server) promoteHTTPS(ln net.Listener) {
+	err := http.Serve(ln, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		http.Redirect(w, r, "https://"+r.Host+r.RequestURI, http.StatusMovedPermanently)
```

Comment:
I think StatusFound or something less permanent is safer cache-wise? I always have to look up the differences, so maybe I'm remembering wrong.